### DISCUSSION TITLE
xmlhttprequest error in edge functions

### DISCUSSION THREAD

**Discussion by rbkayz** [supabase/supabase#9516] [Questions]
Hello community, 

I'm trying to write an edge function in deno that calls the sendEmail API using the AWS SDK (using esm.sh to fetch the package). However, running into an error "NetworkingError: xmlhttprequest is not defined". 

Been racking my brain over it. Tried a polyfill import - import "https://deno.land/x/xhr@0.1.0/mod.ts";

But when i add this line, supabase does not let me deploy the function to deno. Throws an error. 

Any suggestions or help will be incredible. Ty ty

**Comment by cohlar** [comment]
Hi,

I encountered the same issue yesterday and fixed it by adding `import 'https://deno.land/x/xhr@0.1.0/mod.ts';` to my edge function.
Hope this helps.
